contracts/proxies/CurrencyConvertor.sol

```diff
@@ -0,0 +1,134 @@
+    limitations under the License.
+*/
+
+pragma solidity ^0.8.0;
+
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { I_ExchangeWrapper } from "../external/I_ExchangeWrapper.sol";
+import { I_StarkwareContract } from "../external/I_StarkwareContracts.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "hardhat/console.sol";
+
+contract CurrencyConvertor {
+    using SafeERC20 for IERC20;
+
+    // ============ State Variables ============
+
+    ERC20 immutable USDC_ADDRESS;
+
+    uint256 immutable USDC_ASSET_TYPE;
+
+    I_StarkwareContract public immutable STARKWARE_CONTRACT;
+
+    // ============ Constructor ============
+
+    constructor(
+        address starkwareContractAddress,
+        ERC20 usdcAddress,
+        uint256 usdcAssetType
+    )
+    {
+        STARKWARE_CONTRACT = I_StarkwareContract(starkwareContractAddress);
+        USDC_ADDRESS = usdcAddress;
+        USDC_ASSET_TYPE = usdcAssetType;
+    }
+
+  // ============ Events ============
+
+      event LogConvertedDeposit(
+        address indexed account,
+        address source,
+        address exchangeWrapper,
+        ERC20 tokenFrom,
+        uint256 tokenFromAmount,
+        uint256 tokenToAmount
+    );
+
+  // ============ State-Changing Functions ============
+
+  /**
+    * @notice Make a deposit to the Starkware Layer2 Solution, after converting funds to USDC.
+    *  Funds will be withdrawn from the sender and deposited into the StarkWare Layer2 to the starkKey.
+    * @dev Emits LogConvertedDeposit event.
+    *
+    * @param  tokenFrom        The token to convert from.
+    * @param  tokenFromAmount  The amount of `tokenFrom` tokens to deposit.
+    * @param  exchangeWrapper  The ExchangeWrapper contract to trade with.
+    * @param  starkKey         The starkKey of the L2 account to deposit into.
+    * @param  positionId       The positionId of the L2 account to deposit into.
+    * @param  data             Trade parameters for the ExchangeWrapper.
+    */
+  function deposit(
+    ERC20 tokenFrom,
+    uint256 tokenFromAmount,
+    address exchangeWrapper,
+    uint256 starkKey,
+    uint256 positionId,
+    bytes calldata data
+  )
+    external
+    returns (uint256)
+  {
+
+    // Send fromToken to the ExchangeWrapper.
+    IERC20(tokenFrom).safeTransferFrom(
+      msg.sender,
+      exchangeWrapper,
+      tokenFromAmount
+    );
+
+    address self = address(this);
+
+    // Convert fromToken to toToken on the ExchangeWrapper.
+    I_ExchangeWrapper exchangeWrapperContract = I_ExchangeWrapper(exchangeWrapper);
+
+    uint256 tokenToAmount = exchangeWrapperContract.exchange(
+        msg.sender,
+        self,
+        USDC_ADDRESS,
+        tokenFrom,
+        tokenFromAmount,
+        data
+    );
+
+    // Receive toToken from the ExchangeWrapper.
+    // IERC20(USDC_ADDRESS).safeTransferFrom(
```

Comment:
If [these](https://protocol.0x.org/en/latest/advanced/erc20_transformations.html#erc20-transformations) are the docs you're referencing, I don't think the exchange wrapper will be sent the tokens. It seems like in the parameter comments that the output tokens will be transferred to `msg.sender`, but I might be reading something wrong..

```
/// @dev Executes a series of transformations to convert an ERC20 `inputToken`
///      to an ERC20 `outputToken`.
/// @param inputToken The token being provided by the sender.
///        If `0xeee...`, ETH is implied and should be provided with the call.`
/// @param outputToken The token to be acquired by the sender.
///        `0xeee...` implies ETH.

```

Reply:
will be changed for the actual exchange wrapper